genesyscloud/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,218 @@
+	resources := make(ResourceIDMetaMap)
+	archAPI := platformclientv2.NewArchitectApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		schedules, _, getErr := archAPI.GetArchitectSchedules(pageNum, 100, "", "", "", nil)
+		if getErr != nil {
+			return nil, diag.Errorf("Failed to get page of schedules: %v", getErr)
+		}
+
+		if schedules.Entities == nil || len(*schedules.Entities) == 0 {
+			break
+		}
+
+		for _, schedule := range *schedules.Entities {
+			resources[*schedule.Id] = &ResourceMeta{Name: *schedule.Name}
+		}
+	}
+
+	return resources, nil
+}
+
+func architectSchedulesExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllArchitectSchedules),
+		RefAttrs:         map[string]*RefAttrSettings{}, // No references
+	}
+}
+
+func resourceArchitectSchedules() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Architect Schedules",
+
+		CreateContext: createWithPooledClient(createArchitectSchedules),
+		ReadContext:   readWithPooledClient(readArchitectSchedules),
+		UpdateContext: updateWithPooledClient(updateArchitectSchedules),
+		DeleteContext: deleteWithPooledClient(deleteArchitectSchedules),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Name of the schedule.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the schedule.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+            "start": {
+				Description: "Date time is represented as an ISO-8601 string without a timezone.",
+				Type:         schema.TypeString,
+				Required:     true,
+			},
+            "end": {
+				Description: "Date time is represented as an ISO-8601 string without a timezone.",
+				Type:         schema.TypeString,
+				Required:     true,
+			},
+            "rrule": {
+				Description: "An iCal Recurrence Rule (RRULE) string.",
+				Type:         schema.TypeString,
+				Required:     true,
+			},
+		},
+	}
+}
+
+func createArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	start := d.Get("start").(string)
+	end := d.Get("end").(string)
+	rrule := d.Get("rrule").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	archAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	schedStart, err := time.Parse("2006-01-02T15:04:05.000000", start)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", start, err)
+	}
+
+	schedEnd, err := time.Parse("2006-01-02T15:04:05.000000", end)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", end, err)
+	}
+
+	sched := platformclientv2.Schedule{
+		Name:       		&name,
+		Description:      	&description,
```

Comment:
Since description is optional you should only set it if it's an empty string. This is especially important for the update function